src/main/java/com/odysseusinc/arachne/portal/api/v1/controller/BaseStudyController.java

```diff
@@ -172,14 +181,39 @@ public BaseStudyController(BaseStudyService<T, DS, SS, SU> studyService,
                 T study = convert(studyDTO);
                 study = studyService.create(user, study);
                 result = new JsonResult<>(NO_ERROR);
-                result.setResult(convert(study));
+                result.setResult(convertStudyToStudyDTO(study));
             }
         } else {
             result = new JsonResult<>(PERMISSION_DENIED);
         }
         return result;
     }
 
+    @ApiOperation("Get existing or create new workspace.")
+    @RequestMapping(value = "/api/v1/workspace", method = GET)
+    public JsonResult<WD> getOrCreateWorkspace(Principal principal) throws PermissionDeniedException {
+
+        JsonResult<WD> result;
+        IUser user = getUser(principal);
+        T workspace = studyService.findWorkspaceForUser(user.getId());
+        if (workspace == null) {
+            workspace = studyService.createWorkspace(user);
+        }
+        result = new JsonResult<>(NO_ERROR);
+        result.setResult(convertStudyToWorkspaceDTO(workspace));
+        return result;
+    }
+
+    @ApiOperation("Get workspace for specific user.")
+    @RequestMapping(value = "/api/v1/workspace/{userUuid}", method = GET)
+    public JsonResult<WD> getWorkspaceForUser(@PathVariable("userUuid") String userUuid) {
+
+        JsonResult<WD> result = new JsonResult<>(NO_ERROR);
+        WD workspaceDTO = convertStudyToWorkspaceDTO(studyService.findWorkspaceForUser(UserIdUtils.uuidToId(userUuid)));
+        result.setResult(workspaceDTO);
```

Comment:
Oops, and what if workspace didn't found? In this case we should return 404 error to UI, am I right?